log/logger.go

```diff
@@ -129,30 +129,15 @@ func mergeSets(a, b attribute.Set) attribute.Set {
 // WithInstrumentationAttributes returns a [LoggerOption] that sets the
 // instrumentation attributes of a [Logger].
 //
-// The passed attributes will be de-duplicated.
-//
-// Note that [WithInstrumentationAttributeSet] is recommended as
-// it is more efficient and also allows safely reusing the passed argument.
+// This is equivalent to calling WithInstrumentationAttributeSet with an
```

Comment:
```
// Note that the passed attributes are de-duplicated by WithInstrumentationAttributes.
// This is equivalent to calling [WithInstrumentationAttributeSet] with an
```

However, don't you want to clone the passed attributes?
https://github.com/open-telemetry/opentelemetry-go/pull/7266#discussion_r2316507901
If I remember correctly this should solve https://github.com/open-telemetry/opentelemetry-go/pull/7266#discussion_r2307746112

Reply:
Yes, I just wasn't sure it would be acceptable and it wasn't necessary to fix the bug that shows up in the collector/otelzap. I've added it in dbb9d49d2c7fb88d98bc38d302cd73a234f17a46